contracts/potlock/src/potlock_interactions.rs

```diff
@@ -0,0 +1,68 @@
+use crate::potlock_setup;
+use crate::potlock_storage::{self, Pot, Project};
+use crate::potlock_storage::{PotlockId, ProjectId};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[multiversx_sc::module]
+pub trait PotlockInteractions:
+    potlock_storage::PotlockStorage
+    + potlock_setup::PotlockSetup
+    + multiversx_sc_modules::only_admin::OnlyAdminModule
+{
+    #[payable("*")]
+    #[endpoint(addPot)]
+    fn add_pot(&self, name: ManagedBuffer, description: ManagedBuffer) {
+        let payment_for_adding_pot = self.call_value().single_esdt();
+        require!(
+            self.fee_token_identifier().get() == payment_for_adding_pot.token_identifier,
+            "Wrong token identifier for creating a pot!"
+        );
+        require!(
+            self.fee_amount().get() == payment_for_adding_pot.amount,
+            "Wrong fee amount for creating a pot"
+        );
+        let caller = self.blockchain().get_caller();
+
+        let potlock_id = self.potlocks().len() + 1;
+        let potlock = Pot::new(potlock_id, name, description);
+        self.potlocks().push(&potlock);
+
+        self.fee_pot_proposer(potlock_id).set(caller);
+    }
+
+    #[endpoint(applyForPot)]
+    fn apply_for_pot(
+        &self,
+        potlock_id: PotlockId,
+        project_name: ManagedBuffer,
+        description: ManagedBuffer,
+    ) {
+        let project_id = self.projects().len() + 1;
+        let owner = self.blockchain().get_caller();
+        let project = Project::new(project_id, potlock_id, project_name, description, owner);
+        self.projects().push(&project);
+    }
+
+    #[payable("*")]
+    #[endpoint(donateToPot)]
+    fn donate_to_pot(&self, potlock_id: PotlockId) {
+        let payment = self.call_value().single_esdt();
+        let caller = self.blockchain().get_caller();
+        self.require_potlock_exists(potlock_id);
+        self.require_potlock_is_active(potlock_id);
+        self.pot_donations(potlock_id).insert(caller, payment);
```

Comment:
Accepted multiple payments but only with the same token_id.

Reply:
This function is still the same. The payment token is not checked. And in case you only ask for a specific payment token, you would still need a check, to see if there is already some donation from the caller. If it isn't, then you add it, if there's already something, you accumulate it. But this works only if the accepted token_id is always the same and not changeable.

Reply:
Now I see you did this update for the donateToProject endpoint, but it is needed here also.